com.unity.mobile.notifications/Runtime/iOS/iOSNotificationsWrapper.cs

```diff
@@ -70,20 +75,30 @@ internal class iOSNotificationsWrapper : MonoBehaviour
         [DllImport("__Internal")]
         private static extern void _FreeUnmanagediOSNotificationData(IntPtr ptr);
 
-        private delegate void AuthorizationRequestCallback(IntPtr authdata);
+        private delegate void AuthorizationRequestCallback(IntPtr request, iOSAuthorizationRequestData data);
         private delegate void NotificationReceivedCallback(IntPtr notificationData);
 
 #if UNITY_IOS && !UNITY_EDITOR
-        private static AuthorizationRequestCallback s_OnAuthenticationRequestFinished = null;
         private static NotificationReceivedCallback s_OnNotificationReceived = null;
         private static NotificationReceivedCallback s_OnRemoteNotificationReceived = null;
 #endif
 
         public static void RegisterAuthorizationRequestCallback()
         {
 #if UNITY_IOS && !UNITY_EDITOR
-            s_OnAuthenticationRequestFinished = new AuthorizationRequestCallback(AuthorizationRequestReceived);
-            _SetAuthorizationRequestReceivedDelegate(s_OnAuthenticationRequestFinished);
+    #if DEVELOPMENT_BUILD
+            {
+                var nativeSize = _NativeSizeof_iOSNotificationAuthorizationData();
+                var managedSize = Marshal.SizeOf(typeof(iOSAuthorizationRequestData));
+                if (nativeSize != managedSize)
+                {
+                    var error = string.Format("Native/managed struct size missmatch: {0} vs {1}", nativeSize, managedSize);
+                    Debug.LogError(error);
```

Comment:
looks like overkill, logerror and throw exception.

Reply:
Only throw exception?
This is a critical bug, if size doesn't match.

Reply:
yes, only throw exception, since it will print the issue to log already, so no point in printing the same with Debug.LogException

Reply:
Fixed.